auth-web/src/components/datatable/BaseVDataTable.vue

```diff
@@ -61,7 +61,10 @@
 
     <!-- Items -->
     <template v-slot:item="{ item, index }">
-      <tr class="base-table__item-row" :key="item[itemKey]">
+      <tr
+        :key="index*(highlightIndex+2)"
```

Comment:
Sometimes the component doesn't re-render when `highlightIndex` changes. To force it updates, I make key change at the same time so that the component can be refreshed every time.

I use `highlightIndex+2` to keep non-zero number so that it avoids duplicate key warnings.

Reply:
Yes, "key" is how the component knows whether to re-render items.

It sounds like your fix will ensure that all items have unique keys, including the new "first item" (where key=0).

Another thing that might work is to use `:key="Math.random()" so they're always different, but this might re-render the entire table when an item is added/changed (due to reactivity), and ideally you would only add/update the new item.

In short, I think this is OK, but could you add a comment above this to describe what/why you're doing this?

Reply:
To follow up my comments above, the thing I did also make vue re-render the entire table because every key will change when `highlightIndex` changes. I've tested `:key="Math.random()"` also works. Do you think I had better use `Math.random()` instead? 👀

Reply:
Haha, you exactly understand what's going on -- awesome 👍 

Is there any way to assign a new key to an item that is inserted at the top of the table?

Reply:
Ideally we wouldn't re-render the entire table when a new item is added.

Perhaps the keys could be assigned in code not in script?

Reply:
What does it mean, in code not in script? :frowning:

Reply:
In the data array itself, where it can be done once for each item, instead of in script, where it's computed per item every time it re-renders.

Maybe in script is OK, if the list is rendered just once. This will need testing because I don't know off-hand whether the entire list is always re-rendered, or new/changed items are inserted. I'd be OK with you using the random function if you want to try it.

Reply:
I'll keep my solution if there's no better way, considering that random function makes the key unstable and unpredictable.

Reply:
Is there any way to not re-render the entire table?

Reply:
Where do we stand with this?

Reply:
@chenhongjing What's the status of this issue/conversation?

Reply:
Resolved!